# Import Dependencies

In [2]:
import numpy as np
import pandas as pd

pd.set_option('display.max_columns', 500)

# Read in Data

In [3]:
# Columns to keep for the data we want to use
keep_cols= ["leaidC","leaname","fips","stateabb","grade","year","subject",
          "mn_all","mn_asn","mn_blk","mn_fem","mn_hsp","mn_mal","mn_wht",
          "mn_wag","mn_wbg","mn_whg","mn_mfg"
]

In [4]:
long_CS= pd.read_csv("resources\datasets\SEDA_geodist_long_CS_v21.csv")

In [5]:
long_gcs= pd.read_csv("resources\datasets\SEDA_geodist_long_GCS_v21.csv")

In [6]:
# This is the dataset we'll be using
long_naep= pd.read_csv("resources\datasets\SEDA_geodist_long_NAEP_v21.csv")
long_naep= long_naep.loc[:,keep_cols]

In [7]:
long_state= pd.read_csv("resources\datasets\SEDA_geodist_long_State_v21.csv")

In [8]:
pool_cs= pd.read_csv("resources\datasets\SEDA_geodist_pool_CS_v21.csv")

In [9]:
poolsub_cs= pd.read_csv("resources\datasets\SEDA_geodist_poolsub_CS_v21.csv")

# Columns in the data

In [10]:
print(len(long_CS.columns))
print(len(long_gcs.columns))
print(len(long_naep.columns))
print(len(long_state.columns))
print(len(pool_cs.columns))
print(len(poolsub_cs.columns))

50
50
18
50
33
44


In [11]:
columns_df= pd.DataFrame({
    "long_CS": pd.Series(long_CS.columns),
    "long_gcs": pd.Series(long_gcs.columns),
    "long_naep": pd.Series(long_naep.columns),
    "long_state": pd.Series(long_state.columns),
    "pool_cs": pd.Series(pool_cs.columns),
    "poolsub_cs": pd.Series(poolsub_cs.columns)
})

In [12]:
columns_df

long_CS    long_gcs long_naep  long_state       pool_cs  \
0       leaidC      leaidC    leaidC      leaidC        leaidC   
1      leaname     leaname   leaname     leaname       leaname   
2         fips        fips      fips        fips          fips   
3     stateabb    stateabb  stateabb    stateabb      stateabb   
4        grade       grade     grade       grade      subgroup   
5         year        year      year        year       gap_est   
6      subject     subject   subject     subject     tot_asmts   
7   totgyb_all  totgyb_all    mn_all  totgyb_all     cellcount   
8       mn_all      mn_all    mn_asn      mn_all      mn_asmts   
9    mn_all_se   mn_all_se    mn_blk   mn_all_se     mn_avg_ol   
10      sd_all      sd_all    mn_fem      sd_all     mn_grd_ol   
11   sd_all_se   sd_all_se    mn_hsp   sd_all_se     mn_mth_ol   
12  totgyb_asn  totgyb_asn    mn_mal  totgyb_asn     mn_avg_eb   
13      mn_asn      mn_asn    mn_wht      mn_asn     mn_grd_eb   
14   mn_asn_se   mn_asn_se    mn_wag   mn_asn_se     mn_mth_eb   
15      sd_asn      sd_asn    mn_wbg      sd_asn  mn_avg_eb_se   
16   sd_asn_se   sd_asn_se    mn_whg   sd_asn_se  mn_grd_eb_se   
17  totgyb_blk  totgyb_blk    mn_mfg  totgyb_blk  mn_mth_eb_se   
18      mn_blk      mn_blk       NaN      mn_blk  mn_avg_ol_se   
19   mn_blk_se   mn_blk_se       NaN   mn_blk_se  mn_grd_ol_se   
20      sd_blk      sd_blk       NaN      sd_blk  mn_mth_ol_se   
21   sd_blk_se   sd_blk_se       NaN   sd_blk_se     sd_avg_ol   
22  totgyb_fem  totgyb_fem       NaN  totgyb_fem     sd_grd_ol   
23      mn_fem      mn_fem       NaN      mn_fem     sd_mth_ol   
24   mn_fem_se   mn_fem_se       NaN   mn_fem_se     sd_avg_eb   
25      sd_fem      sd_fem       NaN      sd_fem     sd_grd_eb   
26   sd_fem_se   sd_fem_se       NaN   sd_fem_se     sd_mth_eb   
27  totgyb_hsp  totgyb_hsp       NaN  totgyb_hsp  sd_avg_eb_se   
28      mn_hsp      mn_hsp       NaN      mn_hsp  sd_grd_eb_se   
29   mn_hsp_se   mn_hsp_se       NaN   mn_hsp_se  sd_mth_eb_se   
30      sd_hsp      sd_hsp       NaN      sd_hsp  sd_avg_ol_se   
31   sd_hsp_se   sd_hsp_se       NaN   sd_hsp_se  sd_grd_ol_se   
32  totgyb_mal  totgyb_mal       NaN  totgyb_mal  sd_mth_ol_se   
33      mn_mal      mn_mal       NaN      mn_mal           NaN   
34   mn_mal_se   mn_mal_se       NaN   mn_mal_se           NaN   
35      sd_mal      sd_mal       NaN      sd_mal           NaN   
36   sd_mal_se   sd_mal_se       NaN   sd_mal_se           NaN   
37  totgyb_wht  totgyb_wht       NaN  totgyb_wht           NaN   
38      mn_wht      mn_wht       NaN      mn_wht           NaN   
39   mn_wht_se   mn_wht_se       NaN   mn_wht_se           NaN   
40      sd_wht      sd_wht       NaN      sd_wht           NaN   
41   sd_wht_se   sd_wht_se       NaN   sd_wht_se           NaN   
42      mn_wag      mn_wag       NaN      mn_wag           NaN   
43   mn_wag_se   mn_wag_se       NaN   mn_wag_se           NaN   
44      mn_wbg      mn_wbg       NaN      mn_wbg           NaN   
45   mn_wbg_se   mn_wbg_se       NaN   mn_wbg_se           NaN   
46      mn_whg      mn_whg       NaN      mn_whg           NaN   
47   mn_whg_se   mn_whg_se       NaN   mn_whg_se           NaN   
48      mn_mfg      mn_mfg       NaN      mn_mfg           NaN   
49   mn_mfg_se   mn_mfg_se       NaN   mn_mfg_se           NaN   

          poolsub_cs  
0             leaidC  
1            leaname  
2               fips  
3           stateabb  
4           subgroup  
5            gap_est  
6      tot_asmts_ela  
7      tot_asmts_mth  
8      cellcount_ela  
9      cellcount_mth  
10      mn_asmts_ela  
11      mn_asmts_mth  
12     mn_avg_mth_ol  
13     mn_avg_ela_ol  
14     mn_grd_mth_ol  
15     mn_grd_ela_ol  
16     mn_avg_mth_eb  
17     mn_avg_ela_eb  
18     mn_grd_mth_eb  
19     mn_grd_ela_eb  
20  mn_avg_mth_eb_se  
21  mn_avg_ela_eb_se  
22  mn_grd_mth_eb_se  
23  mn_grd_ela_eb_se  
24  mn_avg_mth_ol_se  
25  mn_grd_mth_ol_se  
26  mn_avg_ela_ol_s

# Off with their heads!

In [13]:
long_CS.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
1  100002  Alabama Youth Services     1       AL      8  2009    math   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   

   totgyb_all    mn_all  mn_all_se    sd_all  sd_all_se  totgyb_asn  mn_asn  \
0        29.0 -1.569190   0.209354  0.931462   0.016600         NaN     NaN   
1        36.0 -1.761643   0.206164  0.892374   0.021631         NaN     NaN   
2        29.0 -0.962804   0.203069  0.905344   0.020444         NaN     NaN   
3        27.0 -1.641117   0.223450  0.857457   0.018269         NaN     NaN   
4        33.0 -1.069076   0.190945  0.913058   0.014726         NaN     NaN   

   mn_asn_se  sd_asn  sd_asn_se  totgyb_blk    mn_blk  mn_blk_se    sd_blk  \
0        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
1        NaN     NaN        NaN        20.0 -1.986859   0.284851  0.823336   
2        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
3        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
4        NaN     NaN        NaN         NaN       NaN        NaN       NaN   

   sd_blk_se  totgyb_fem  mn_fem  mn_fem_se  sd_fem  sd_fem_se  totgyb_hsp  \
0        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
1   0.022021         NaN     NaN        NaN     NaN        NaN         NaN   
2        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
3        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
4        NaN         NaN     NaN        NaN     NaN        NaN         NaN   

   mn_hsp  mn_hsp_se  sd_hsp  sd_hsp_se  totgyb_mal    mn_mal  mn_mal_se  \
0     NaN        NaN     NaN        NaN        26.0 -1.631368   0.224605   
1     NaN        NaN     NaN        NaN        33.0 -1.814169   0.223662   
2     NaN        NaN     NaN        NaN        27.0 -0.990942   0.211828   
3     NaN        NaN     NaN        NaN        26.0 -1.583394   0.238398   
4     NaN        NaN     NaN        NaN        33.0 -1.134681   0.190324   

     sd_mal  sd_mal_se  totgyb_wht  mn_wht  mn_wht_se  sd_wht  sd_wht_se  \
0  0.948279   0.018422         NaN     NaN        NaN     NaN        NaN   
1  0.944506   0.024233         NaN     NaN        NaN     NaN        NaN   
2  0.912120   0.021901         NaN     NaN        NaN     NaN        NaN   
3  0.908649   0.020670         NaN     NaN        NaN     NaN        NaN   
4  0.908174   0.016259         NaN     NaN        NaN     NaN        NaN   

   mn_wag  mn_wag_se  mn_wbg  mn_wbg_se  mn_whg  mn_whg_se  mn_mfg  mn_mfg_se  
0     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
1     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
2     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
3     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
4     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN

In [14]:
long_gcs.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
1  100002  Alabama Youth Services     1       AL      8  2009    math   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   

   totgyb_all    mn_all  mn_all_se    sd_all  sd_all_se  totgyb_asn  mn_asn  \
0        29.0  3.364652   0.618425  2.751515   0.049036         NaN     NaN   
1        36.0  2.138392   0.685979  2.969243   0.071975         NaN     NaN   
2        29.0  5.155901   0.599859  2.674363   0.060393         NaN     NaN   
3        27.0  2.539424   0.743498  2.853061   0.060789         NaN     NaN   
4        33.0  4.841975   0.564046  2.697150   0.043501         NaN     NaN   

   mn_asn_se  sd_asn  sd_asn_se  totgyb_blk    mn_blk  mn_blk_se    sd_blk  \
0        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
1        NaN     NaN        NaN        20.0  1.389016   0.947799  2.739531   
2        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
3        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
4        NaN     NaN        NaN         NaN       NaN        NaN       NaN   

   sd_blk_se  totgyb_fem  mn_fem  mn_fem_se  sd_fem  sd_fem_se  totgyb_hsp  \
0        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
1   0.073273         NaN     NaN        NaN     NaN        NaN         NaN   
2        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
3        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
4        NaN         NaN     NaN        NaN     NaN        NaN         NaN   

   mn_hsp  mn_hsp_se  sd_hsp  sd_hsp_se  totgyb_mal    mn_mal  mn_mal_se  \
0     NaN        NaN     NaN        NaN        26.0  3.180978   0.663477   
1     NaN        NaN     NaN        NaN        33.0  1.963618   0.744201   
2     NaN        NaN     NaN        NaN        27.0  5.072781   0.625734   
3     NaN        NaN     NaN        NaN        26.0  2.731487   0.793233   
4     NaN        NaN     NaN        NaN        33.0  4.648181   0.562213   

     sd_mal  sd_mal_se  totgyb_wht  mn_wht  mn_wht_se  sd_wht  sd_wht_se  \
0  2.801194   0.054420         NaN     NaN        NaN     NaN        NaN   
1  3.142704   0.080630         NaN     NaN        NaN     NaN        NaN   
2  2.694380   0.064694         NaN     NaN        NaN     NaN        NaN   
3  3.023397   0.068775         NaN     NaN        NaN     NaN        NaN   
4  2.682725   0.048029         NaN     NaN        NaN     NaN        NaN   

   mn_wag  mn_wag_se  mn_wbg  mn_wbg_se  mn_whg  mn_whg_se  mn_mfg  mn_mfg_se  
0     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
1     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
2     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
3     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
4     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN

In [15]:
long_naep.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
1  100002  Alabama Youth Services     1       AL      8  2009    math   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   

      mn_all  mn_asn     mn_blk  mn_fem  mn_hsp     mn_mal  mn_wht  mn_wag  \
0  209.40759     NaN        NaN     NaN     NaN  207.21269     NaN     NaN   
1  217.34308     NaN  208.98753     NaN     NaN  215.39435     NaN     NaN   
2  230.81302     NaN        NaN     NaN     NaN  229.81973     NaN     NaN   
3  221.81458     NaN        NaN     NaN     NaN  223.95609     NaN     NaN   
4  227.06160     NaN        NaN     NaN     NaN  224.74576     NaN     NaN   

   mn_wbg  mn_whg  mn_mfg  
0     NaN     NaN     NaN  
1     NaN     NaN     NaN  
2     NaN     NaN     NaN  
3     NaN     NaN     NaN  
4     NaN     NaN     NaN

In [16]:
long_state.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
1  100002  Alabama Youth Services     1       AL      8  2009    math   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   

   totgyb_all    mn_all  mn_all_se    sd_all  sd_all_se  totgyb_asn  mn_asn  \
0        29.0 -1.168019   0.190559  0.920672   0.005181         NaN     NaN   
1        36.0 -1.282802   0.190213  0.887760   0.005257         NaN     NaN   
2        29.0 -0.708781   0.188333  0.920245   0.005173         NaN     NaN   
3        27.0 -1.233343   0.215231  0.885889   0.005659         NaN     NaN   
4        33.0 -0.804737   0.179870  0.928212   0.005347         NaN     NaN   

   mn_asn_se  sd_asn  sd_asn_se  totgyb_blk    mn_blk  mn_blk_se    sd_blk  \
0        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
1        NaN     NaN        NaN        20.0 -1.496076   0.265494  0.825772   
2        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
3        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
4        NaN     NaN        NaN         NaN       NaN        NaN       NaN   

   sd_blk_se  totgyb_fem  mn_fem  mn_fem_se  sd_fem  sd_fem_se  totgyb_hsp  \
0        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
1    0.00961         NaN     NaN        NaN     NaN        NaN         NaN   
2        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
3        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
4        NaN         NaN     NaN        NaN     NaN        NaN         NaN   

   mn_hsp  mn_hsp_se  sd_hsp  sd_hsp_se  totgyb_mal    mn_mal  mn_mal_se  \
0     NaN        NaN     NaN        NaN        26.0 -1.225511   0.204771   
1     NaN        NaN     NaN        NaN        33.0 -1.332543   0.207005   
2     NaN        NaN     NaN        NaN        27.0 -0.735563   0.196830   
3     NaN        NaN     NaN        NaN        26.0 -1.176524   0.230360   
4     NaN        NaN     NaN        NaN        33.0 -0.867299   0.179208   

     sd_mal  sd_mal_se  totgyb_wht  mn_wht  mn_wht_se  sd_wht  sd_wht_se  \
0  0.935234   0.008267         NaN     NaN        NaN     NaN        NaN   
1  0.934875   0.009110         NaN     NaN        NaN     NaN        NaN   
2  0.926287   0.008442         NaN     NaN        NaN     NaN        NaN   
3  0.934023   0.009104         NaN     NaN        NaN     NaN        NaN   
4  0.923844   0.008249         NaN     NaN        NaN     NaN        NaN   

   mn_wag  mn_wag_se  mn_wbg  mn_wbg_se  mn_whg  mn_whg_se  mn_mfg  mn_mfg_se  
0     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
1     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
2     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
3     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
4     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN

In [17]:
pool_cs.head()

leaidC                 leaname  fips stateabb subgroup  gap_est  tot_asmts  \
0  100002  Alabama Youth Services     1       AL      all        0        185   
1  100002  Alabama Youth Services     1       AL      blk        0         20   
2  100002  Alabama Youth Services     1       AL      mal        0        176   
3  100005        Albertville City     1       AL      all        0      26185   
4  100005        Albertville City     1       AL      fem        0      11421   

   cellcount    mn_asmts  mn_avg_ol  mn_grd_ol  mn_mth_ol  mn_avg_eb  \
0          6   30.833334        NaN        NaN        NaN        NaN   
1          1   20.000000        NaN        NaN        NaN        NaN   
2          6   29.333334        NaN        NaN        NaN        NaN   
3         82  319.329250  -0.280656  -0.061207  -0.116414  -0.280273   
4         69  165.521740  -0.225336  -0.040484  -0.190132  -0.225179   

   mn_grd_eb  mn_mth_eb  mn_avg_eb_se  mn_grd_eb_se  mn_mth_eb_se  \
0        NaN        NaN           NaN           NaN           NaN   
1        NaN        NaN           NaN           NaN           NaN   
2        NaN        NaN           NaN           NaN           NaN   
3  -0.058287  -0.112834      0.014924      0.011383      0.029116   
4  -0.037043  -0.187486      0.016971      0.012768      0.032495   

   mn_avg_ol_se  mn_grd_ol_se  mn_mth_ol_se  sd_avg_ol  sd_grd_ol  sd_mth_ol  \
0           NaN           NaN           NaN        NaN        NaN        NaN   
1           NaN           NaN           NaN        NaN        NaN        NaN   
2           NaN           NaN           NaN        NaN        NaN        NaN   
3      0.014940      0.011783      0.029962   0.845352  -0.010274  -0.019646   
4      0.017021      0.013442      0.033580   0.768277        NaN  -0.020952   

   sd_avg_eb  sd_grd_eb  sd_mth_eb  sd_avg_eb_se  sd_grd_eb_se  sd_mth_eb_se  \
0        NaN        NaN        NaN           NaN           NaN           NaN   
1        NaN        NaN        NaN           NaN           NaN           NaN   
2        NaN        NaN        NaN           NaN           NaN           NaN   
3   0.845379  -0.008705  -0.021662      0.007639      0.005382      0.014190   
4   0.772706        NaN  -0.026816      0.010658           NaN      0.017749   

   sd_avg_ol_se  sd_grd_ol_se  sd_mth_ol_se  
0           NaN           NaN           NaN  
1           NaN           NaN           NaN  
2           NaN           NaN           NaN  
3      0.007737      0.006004      0.015404  
4      0.011180           NaN      0.021319

In [18]:
poolsub_cs.head()

leaidC                 leaname  fips stateabb subgroup  gap_est  \
0  100002  Alabama Youth Services     1       AL      all        0   
1  100002  Alabama Youth Services     1       AL      blk        0   
2  100002  Alabama Youth Services     1       AL      mal        0   
3  100005        Albertville City     1       AL      all        0   
4  100005        Albertville City     1       AL      fem        0   

   tot_asmts_ela  tot_asmts_mth  cellcount_ela  cellcount_mth  mn_asmts_ela  \
0           91.0           94.0            3.0            3.0     30.333334   
1            NaN           20.0            NaN            1.0           NaN   
2           86.0           90.0            3.0            3.0     28.666666   
3        12754.0        13431.0           40.0           42.0    318.850010   
4         5621.0         5800.0           34.0           35.0    165.323530   

   mn_asmts_mth  mn_avg_mth_ol  mn_avg_ela_ol  mn_grd_mth_ol  mn_grd_ela_ol  \
0     31.333334            NaN            NaN            NaN            NaN   
1     20.000000            NaN            NaN            NaN            NaN   
2     30.000000            NaN            NaN            NaN            NaN   
3    319.785710      -0.339745      -0.222717      -0.060150      -0.061749   
4    165.714280      -0.323230      -0.129925      -0.032657      -0.048354   

   mn_avg_mth_eb  mn_avg_ela_eb  mn_grd_mth_eb  mn_grd_ela_eb  \
0            NaN            NaN            NaN            NaN   
1            NaN            NaN            NaN            NaN   
2            NaN            NaN            NaN            NaN   
3      -0.337170      -0.223722      -0.061004      -0.054038   
4      -0.319693      -0.132169      -0.036577      -0.034901   

   mn_avg_mth_eb_se  mn_avg_ela_eb_se  mn_grd_mth_eb_se  mn_grd_ela_eb_se  \
0               NaN               NaN               NaN               NaN   
1               NaN               NaN               NaN               NaN   
2               NaN               NaN               NaN               NaN   
3          0.020049          0.020477          0.014545          0.014061   
4          0.022668          0.023309          0.016292          0.015155   

   mn_avg_mth_ol_se  mn_grd_mth_ol_se  mn_avg_ela_ol_se  mn_grd_ela_ol_se  \
0               NaN               NaN               NaN               NaN   
1               NaN               NaN               NaN               NaN   
2               NaN               NaN               NaN               NaN   
3          0.020326          0.015804          0.020785          0.016567   
4          0.023162          0.018212          0.023779          0.018764   

   sd_avg_mth_ol  sd_avg_ela_ol  sd_grd_mth_ol  sd_grd_ela_ol  sd_avg_mth_eb  \
0            NaN            NaN            NaN            NaN            NaN   
1            NaN            NaN            NaN            NaN            NaN   
2            NaN            NaN            NaN            NaN            NaN   
3       0.846394       0.858098      -0.030082            NaN       0.842975   
4       0.770740       0.775054            NaN            NaN       0.764392   

   sd_avg_ela_eb  sd_grd_mth_eb  sd_grd_ela_eb  sd_avg_mth_eb_se  \
0            NaN            NaN            NaN               NaN   
1            NaN            NaN            NaN               NaN   
2            NaN            NaN            NaN               NaN   
3       0.858925      -0.021912            NaN          0.010473   
4       0.785076            NaN            NaN          0.013753   

   sd_avg_ela_eb_se  sd_grd_mth_eb_se  sd_grd_ela_eb_se  sd_avg_mth_ol_se  \
0               NaN               NaN               NaN               NaN   
1               NaN               NaN               NaN               NaN   
2               NaN               NaN               NaN               NaN   
3          0.010814           0.00755               NaN          0.010886   
4          0.014301               NaN   

# Describe the datasets

In [19]:
desc_long_CS= long_CS.describe()
desc_long_gcs= long_gcs.describe()
desc_long_naep= long_naep.describe()
desc_long_state= long_state.describe()
desc_pool_cs= pool_cs.describe()
desc_poolsub_cs= poolsub_cs.describe()

In [20]:
desc_long_CS

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

          totgyb_all         mn_all      mn_all_se         sd_all  \
count  825814.000000  825814.000000  825814.000000  825814.000000   
mean      328.138101       0.045220       0.112220       0.876175   
std      1166.280140       0.400055       0.054619       0.121183   
min        20.000000      -3.186861       0.010884       0.042187   
25%        55.000000      -0.205949       0.070441       0.807569   
50%       119.000000       0.049993       0.099618       0.873354   
75%       277.000000       0.300706       0.149012       0.940343   
max     76800.000000       3.804560       1.841295       4.186017   

           sd_all_se    totgyb_asn        mn_asn     mn_asn_se        sd_asn  \
count  825814.000000  85523.000000  85523.000000  85523.000000  85523.000000   
mean        0.069390    122.689382      0.608329      0.190778      0.987328   
std         0.044821    483.938421      0.556105      0.085566      0.158555   
min         0.007450     20.000000     -3.400659      0.015841      0.177361   
25%         0.029866     29.000000      0.254931      0.131710      0.904609   
50%         0.066962     47.000000      0.610957      0.190276      0.975054   
75%         0.097356     97.000000      0.965965      0.235567      1.049185   
max         1.416418  12214.000000      4.290233      2.123441      3.989972   

          sd_asn_se     totgyb_blk         mn_blk      mn_blk_se  \
count  85523.000000  183171.000000  183171.000000  183171.000000   
mean       0.084771     224.051460      -0.438864       0.132186   
std        0.077507     719.344149       0.326138       0.058851   
min        0.011620      20.000000      -2.802191       0.015411   
25%        0.029321      36.000000      -0.654225       0.083788   
50%        0.053963      72.000000      -0.444535       0.125113   
75%        0.121298     170.000000      -0.225432       0.177508   
max        1.861681   22678.000000       1.476525       1.171951   

              sd_blk      sd_blk_se     totgyb_fem         mn_fem  \
count  183171.000000  183171.000000  684574.000000  684574.000000   
mean        0.849140       0.065292     188.532721       0.120130   
std         0.110419       0.049079     621.838991       0.403455   
min         0.103305       0.009407      20.000000      -2.405667   
25%         0.791534       0.021645      39.000000      -0.140214   
50%         0.844292       0.057352      74.000000       0.121420   
75%         0.906163       0.096868     159.000000       0.379138   
max         2.570137       1.111087   37799.000000       3.931363   

           mn_fem_se         sd_fem      sd_fem_se     totgyb_hsp  \
count  684574.000000  684574.000000  684574.000000  222124.000000   
mean        0.130864       0.855555       0.067869     252.636262   
std         0.055945       0.115374       0.049798    1028.612933   
min         0.014748       0.105092       0.007860      20.000000   
25%         0.087132       0.796186       0.018935      33.000000   
50%         0.122946       0.850554       0.065888      66.000000   
75%         0.171063       0.911051       0.101516     177.000000   
max         1.691966       3.085215       1.206432   38791.000000   

              mn_hsp      mn_hsp_se         sd_hsp      sd_hsp_se  \
count  222124.000000  222124.000000  222124.000000  222124.000000   
mean       -0.278085       0.135376       0.860349       0.060

In [21]:
desc_long_gcs

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

          totgyb_all         mn_all      mn_all_se         sd_all  \
count  825814.000000  825814.000000  825814.000000  825814.000000   
mean      328.138101       5.565391       0.336394       2.628670   
std      1166.280140       1.988355       0.165724       0.413478   
min        20.000000      -4.601914       0.032604       0.140370   
25%        55.000000       4.086436       0.209938       2.364256   
50%       119.000000       5.516290       0.297787       2.632125   
75%       277.000000       7.023564       0.442947       2.871308   
max     76800.000000      16.319139       5.515627      12.365393   

           sd_all_se    totgyb_asn        mn_asn     mn_asn_se        sd_asn  \
count  825814.000000  85523.000000  85523.000000  85523.000000  85523.000000   
mean        0.208084    122.689382      7.200015      0.569769      2.953043   
std         0.135414    483.938421      2.442277      0.256700      0.515028   
min         0.022315     20.000000     -5.865470      0.047452      0.523919   
25%         0.090063     29.000000      5.448634      0.393414      2.646907   
50%         0.199901     47.000000      7.109251      0.566381      2.931653   
75%         0.291724     97.000000      8.867107      0.704836      3.185928   
max         3.984298  12214.000000     22.275122      6.696728     13.276047   

          sd_asn_se     totgyb_blk         mn_blk      mn_blk_se  \
count  85523.000000  183171.000000  183171.000000  183171.000000   
mean       0.252941     224.051460       4.132336       0.396487   
std        0.231137     719.344149       1.788857       0.179498   
min        0.034808      20.000000      -3.348936       0.046091   
25%        0.087202      36.000000       2.760144       0.249794   
50%        0.161543      72.000000       4.094108       0.374391   
75%        0.362594     170.000000       5.464153       0.531483   
max        6.194471   22678.000000      12.196367       3.712929   

              sd_blk      sd_blk_se     totgyb_fem         mn_fem  \
count  183171.000000  183171.000000  684574.000000  684574.000000   
mean        2.546893       0.195934     188.532721       5.806853   
std         0.376843       0.148733     621.838991       2.037545   
min         0.276098       0.028026      20.000000      -3.786019   
25%         2.312592       0.064641      39.000000       4.270745   
50%         2.548520       0.170269      74.000000       5.754171   
75%         2.761659       0.290041     159.000000       7.313653   
max         7.592120       3.328277   37799.000000      19.613150   

           mn_fem_se         sd_fem      sd_fem_se     totgyb_hsp  \
count  684574.000000  684574.000000  684574.000000  222124.000000   
mean        0.392497       2.566808       0.203577     252.636262   
std         0.170557       0.394003       0.150433    1028.612933   
min         0.043832       0.318354       0.023545      20.000000   
25%         0.259561       2.319130       0.056467      33.000000   
50%         0.369074       2.568287       0.196127      66.000000   
75%         0.512901       2.786792       0.304254     177.000000   
max         5.161969       9.241808       3.822171   38791.000000   

              mn_hsp      mn_hsp_se         sd_hsp      sd_hsp_se  \
count  222124.000000  222124.000000  222124.000000  222124.000000   
mean        4.514001       0.405153       2.573891       0.180

In [22]:
desc_long_naep

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

              mn_all        mn_asn         mn_blk         mn_fem  \
count  825814.000000  85523.000000  183171.000000  684574.000000   
mean      245.368960    264.213928     228.758834     248.236700   
std        24.817281     30.265167      22.675072      24.501244   
min       114.207150     99.107666     129.180240     123.957090   
25%       228.196378    243.636095     213.230470     230.850810   
50%       245.202560    263.100400     229.219020     247.870575   
75%       262.570257    283.998410     244.799475     265.586045   
max       375.458440    441.867650     314.946560     403.577090   

              mn_hsp         mn_mal         mn_wht        mn_wag  \
count  222124.000000  695482.000000  733460.000000  84024.000000   
mean      233.040101     243.568544     250.108341     -7.432532   
std        24.205590      25.659113      23.724225     14.264807   
min       132.774960     113.350510     130.434510   -131.135770   
25%       216.558828     226.164378     233.195263    -16.002594   
50%       233.525605     243.450965     249.758835     -8.062675   
75%       249.974620     260.939640     266.682310      0.369026   
max       336.939790     377.005740     376.761720    137.154270   

              mn_wbg         mn_whg         mn_mfg  
count  164413.000000  196243.000000  658277.000000  
mean       20.692708      16.047852      -4.591977  
std         9.994392      10.275893       8.664576  
min       -25.843262     -43.132675    -106.446230  
25%        14.246857       9.267326      -9.915894  
50%        20.054749      15.276093      -4.178528  
75%        26.478058      22.158958       1.042725  
max       147.544370     153.679550      58.579193

In [23]:
desc_long_state

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

          totgyb_all         mn_all      mn_all_se         sd_all  \
count  825814.000000  825814.000000  825814.000000  825814.000000   
mean      328.138101       0.002542       0.104061       0.903563   
std      1166.280140       0.340808       0.055142       0.104510   
min        20.000000      -2.972786       0.000000       0.251772   
25%        55.000000      -0.207796       0.062064       0.849491   
50%       119.000000       0.001459       0.092435       0.901879   
75%       277.000000       0.212722       0.142700       0.954235   
max     76800.000000       3.656457       1.760634       3.719599   

           sd_all_se    totgyb_asn        mn_asn     mn_asn_se        sd_asn  \
count  825814.000000  85523.000000  85523.000000  85523.000000  85523.000000   
mean        0.058912    122.689382      0.546172      0.179057      0.991131   
std         0.043337    483.938421      0.514938      0.082941      0.143650   
min         0.000000     20.000000     -3.424887      0.009375      0.296857   
25%         0.021279     29.000000      0.231252      0.121399      0.918437   
50%         0.058466     47.000000      0.555609      0.179640      0.971996   
75%         0.086205     97.000000      0.874363      0.223434      1.045130   
max         1.343618  12214.000000      4.114595      1.951427      3.776745   

          sd_asn_se     totgyb_blk         mn_blk      mn_blk_se  \
count  85523.000000  183171.000000  183171.000000  183171.000000   
mean       0.073129     224.051460      -0.416928       0.122927   
std        0.072831     719.344149       0.293577       0.058248   
min        0.000000      20.000000      -2.543317       0.006473   
25%        0.020539      36.000000      -0.605537       0.075641   
50%        0.046141      72.000000      -0.421108       0.116818   
75%        0.107019     170.000000      -0.230172       0.168060   
max        1.751017   22678.000000       1.738854       1.169030   

              sd_blk      sd_blk_se     totgyb_fem         mn_fem  \
count  183171.000000  183171.000000  684574.000000  684574.000000   
mean        0.876999       0.054904     188.532721       0.074448   
std         0.095191       0.046300     621.838991       0.346849   
min         0.273550       0.000000      20.000000      -2.267843   
25%         0.828387       0.011884      39.000000      -0.146036   
50%         0.875378       0.049883      74.000000       0.072006   
75%         0.921897       0.085082     159.000000       0.292048   
max         2.518226       1.123573   37799.000000       3.609545   

           mn_fem_se         sd_fem      sd_fem_se     totgyb_hsp  \
count  684574.000000  684574.000000  684574.000000  222124.000000   
mean        0.122631       0.884136       0.056920     252.636262   
std         0.055918       0.100978       0.047837    1028.612933   
min         0.005450       0.241442       0.000000      20.000000   
25%         0.079300       0.834420       0.007461      33.000000   
50%         0.115517       0.881488       0.057254      66.000000   
75%         0.163121       0.929520       0.089491     177.000000   
max         1.605439       3.055364       1.159092   38791.000000   

              mn_hsp      mn_hsp_se         sd_hsp      sd_hsp_se  \
count  222124.000000  222124.000000  222124.000000  222124.000000   
mean       -0.269196       0.125265       0.878575       0.049

In [24]:
desc_pool_cs

leaidC          fips       gap_est     tot_asmts     cellcount  \
count  7.214700e+04  72147.000000  72147.000000  7.214700e+04  72147.000000   
mean   2.932015e+06     29.187506      0.257585  1.866353e+04     62.831511   
std    1.520158e+06     15.201987      0.437307  6.298952e+04     24.394492   
min    1.000020e+05      1.000000      0.000000  2.000000e+01      1.000000   
25%    1.734045e+06     17.000000      0.000000  2.408000e+03     56.000000   
50%    2.926850e+06     29.000000      0.000000  6.301000e+03     72.000000   
75%    4.108770e+06     41.000000      1.000000  1.622100e+04     82.000000   
max    5.606240e+06     56.000000      1.000000  4.685211e+06     84.000000   

           mn_asmts     mn_avg_ol     mn_grd_ol     mn_mth_ol     mn_avg_eb  \
count  72147.000000  67235.000000  49956.000000  60872.000000  67242.000000   
mean     269.174043      0.049678     -0.006932      0.017771      0.049536   
std      907.378454      0.369447      0.046488      0.167213      0.364393   
min       20.000000     -1.841032     -0.389979     -0.749475     -1.690200   
25%       41.000000     -0.172902     -0.035998     -0.100292     -0.168033   
50%       92.738098      0.007239     -0.006866      0.013402      0.003202   
75%      235.928570      0.281770      0.022449      0.142542      0.280777   
max    70988.047000      1.907693      0.337531      0.839080      1.755396   

          mn_grd_eb     mn_mth_eb  mn_avg_eb_se  mn_grd_eb_se  mn_mth_eb_se  \
count  49956.000000  60872.000000  67242.000000  49956.000000  60872.000000   
mean      -0.006929      0.017796      0.025079      0.014581      0.037474   
std        0.040103      0.156233      0.014751      0.003376      0.009806   
min       -0.279810     -0.679975      0.004379      0.004757      0.008545   
25%       -0.032037     -0.093153      0.016375      0.011994      0.030515   
50%       -0.006770      0.011162      0.020405      0.013999      0.035813   
75%        0.018468      0.138895      0.027379      0.016665      0.043077   
max        0.226243      0.745145      0.131102      0.024750      0.071296   

       mn_avg_ol_se  mn_grd_ol_se  mn_mth_ol_se     sd_avg_ol     sd_grd_ol  \
count  67235.000000  49956.000000  60872.000000  50458.000000  37565.000000   
mean       0.028555      0.016093      0.041431      0.865058     -0.003462   
std        0.025651      0.004545      0.013068      0.063459      0.013257   
min        0.004398      0.004841      0.008633      0.625352     -0.095699   
25%        0.016478      0.012583      0.031945      0.825000     -0.011145   
50%        0.020713      0.015057      0.038357      0.857771     -0.004035   
75%        0.028598      0.018717      0.048406      0.897884      0.004283   
max        0.225080      0.029539      0.082737      1.546541      0.076853   

          sd_mth_ol     sd_avg_eb     sd_grd_eb     sd_mth_eb  sd_avg_eb_se  \
count  46960.000000  50458.000000  37565.000000  46960.000000  50458.000000   
mean      -0.019809      0.864829     -0.003512     -0.020047      0.007739   
std        0.044403      0.061278      0.010821      0.038944      0.003717   
min       -0.222643      0.636588     -0.079389     -0.168573      0.002448   
25%       -0.051629      0.825744     -0.010255     -0.048239      0.004447   
50%       -0.022158      0.857518     -0.004011     -0.022548      0.007283   
75%        0.008066      0.896905      0.003049      0.004354      0.010421   
max        0.260459      1.497011      0.055841      0.215035      0.026108   

       sd_grd_eb_se  sd_mth_eb_se  sd_avg_ol_se  sd_grd_ol_se  sd_mth_ol_se  
count  37565.000000  46960.000000  50458.000000  37565.000000  46960.000000  
mean       0.004317      0.012381      0.008350      0.004867      0.013663  
std        0.001422      0.004771      0.004879      0.001889      0.005992  
min        0.001912      0.004819      0.002450      0.001940      0.004869  
25%        0.002956      0.007802      0.004493      0.003076  

In [25]:
desc_poolsub_cs

leaidC          fips       gap_est  tot_asmts_ela  tot_asmts_mth  \
count  7.214700e+04  72147.000000  72147.000000   7.204200e+04   7.184800e+04   
mean   2.932015e+06     29.187506      0.257585   9.754440e+03   8.960421e+03   
std    1.520158e+06     15.201987      0.437307   3.322655e+04   3.008000e+04   
min    1.000020e+05      1.000000      0.000000   2.000000e+01   2.000000e+01   
25%    1.734045e+06     17.000000      0.000000   1.243000e+03   1.173000e+03   
50%    2.926850e+06     29.000000      0.000000   3.245000e+03   3.061000e+03   
75%    4.108770e+06     41.000000      1.000000   8.347000e+03   7.846250e+03   
max    5.606240e+06     56.000000      1.000000   2.331930e+06   2.353281e+06   

       cellcount_ela  cellcount_mth  mn_asmts_ela  mn_asmts_mth  \
count   72042.000000   71848.000000  72042.000000  71848.000000   
mean       32.374879      30.630693    269.390770    269.704605   
std        12.340548      12.272864    906.428482    910.487181   
min         1.000000       1.000000     20.000000     20.000000   
25%        29.000000      25.000000     41.000000     41.151516   
50%        36.000000      35.000000     92.857140     93.047623   
75%        42.000000      41.000000    236.472923    236.694332   
max        42.000000      42.000000  70664.547000  71311.547000   

       mn_avg_mth_ol  mn_avg_ela_ol  mn_grd_mth_ol  mn_grd_ela_ol  \
count   66342.000000   66431.000000   46268.000000   39706.000000   
mean        0.059146       0.041812      -0.008441      -0.005112   
std         0.379982       0.377123       0.056426       0.043583   
min        -1.855213      -5.159541      -0.350875      -0.324992   
25%        -0.159971      -0.224729      -0.044345      -0.032920   
50%         0.033562       0.014001      -0.008844      -0.004648   
75%         0.288862       0.286326       0.026452       0.023193   
max         1.968772       2.159821       0.312431       0.195485   

       mn_avg_mth_eb  mn_avg_ela_eb  mn_grd_mth_eb  mn_grd_ela_eb  \
count   66540.000000   66586.000000   46585.000000   39929.000000   
mean        0.058966       0.041122      -0.008556      -0.005266   
std         0.372265       0.370975       0.047500       0.035819   
min        -1.676959      -1.851148      -0.314171      -0.253082   
25%        -0.150619      -0.224407      -0.038295      -0.027976   
50%         0.028197       0.012438      -0.008429      -0.005061   
75%         0.283986       0.285372       0.020850       0.018064   
max         1.948141       1.905320       0.225116       0.145515   

       mn_avg_mth_eb_se  mn_avg_ela_eb_se  mn_grd_mth_eb_se  mn_grd_ela_eb_se  \
count      66540.000000      66586.000000      46585.000000      39929.000000   
mean           0.032353          0.031137          0.018063          0.015136   
std            0.015806          0.014853          0.003873          0.002451   
min            0.005857          0.005965          0.004775          0.005118   
25%            0.022331          0.021663          0.015292          0.013484   
50%            0.027569          0.026806          0.017619          0.015060   
75%            0.036257          0.034893          0.020716          0.016877   
max            0.134209          0.129593          0.028179          0.041197   

       mn_avg_mth_ol_se  mn_grd_mth_ol_se  mn_avg_ela_ol_se  mn_grd_ela_ol_se  \
count      66342.000000      46268.000000      66431.000000      39706.000000   
mean           0.039907          0.021972          0.037100          0.019291   
std            0.032141          0.005625          0.027989          0.003784   
min            0.006052          0.006359          0.006054          0.006579   
25%            0.023087          0.017610          0.022285          0.016276   
50%            0.029216          0.020903          0.028245          0.018917   
75%            0.041123          0.025632          0.038840          0.022212   
max            0.235657          0.036310          0.2219

# Trying to make sense of the data

## NAEP

In [26]:
desc_long_naep

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

              mn_all        mn_asn         mn_blk         mn_fem  \
count  825814.000000  85523.000000  183171.000000  684574.000000   
mean      245.368960    264.213928     228.758834     248.236700   
std        24.817281     30.265167      22.675072      24.501244   
min       114.207150     99.107666     129.180240     123.957090   
25%       228.196378    243.636095     213.230470     230.850810   
50%       245.202560    263.100400     229.219020     247.870575   
75%       262.570257    283.998410     244.799475     265.586045   
max       375.458440    441.867650     314.946560     403.577090   

              mn_hsp         mn_mal         mn_wht        mn_wag  \
count  222124.000000  695482.000000  733460.000000  84024.000000   
mean      233.040101     243.568544     250.108341     -7.432532   
std        24.205590      25.659113      23.724225     14.264807   
min       132.774960     113.350510     130.434510   -131.135770   
25%       216.558828     226.164378     233.195263    -16.002594   
50%       233.525605     243.450965     249.758835     -8.062675   
75%       249.974620     260.939640     266.682310      0.369026   
max       336.939790     377.005740     376.761720    137.154270   

              mn_wbg         mn_whg         mn_mfg  
count  164413.000000  196243.000000  658277.000000  
mean       20.692708      16.047852      -4.591977  
std         9.994392      10.275893       8.664576  
min       -25.843262     -43.132675    -106.446230  
25%        14.246857       9.267326      -9.915894  
50%        20.054749      15.276093      -4.178528  
75%        26.478058      22.158958       1.042725  
max       147.544370     153.679550      58.579193

The NAEP score is derived from the [National Assessment of Educational Progress](https://nces.ed.gov/nationsreportcard/ "NAEP Nations Report Card"). The test is on a 0-500 scale, which we can see reflected in the means of all the schools *(mn_all)*.

In [27]:
print(f'Minimum AVERAGE score for all schools: {desc_long_naep.loc["min","mn_all"]}')
print(f'Maximum AVERAGE score for all schools: {desc_long_naep.loc["max","mn_all"]}')

Minimum AVERAGE score for all schools: 114.20715
Maximum AVERAGE score for all schools: 375.45844


The above shows that the AVERAGE maximum score for any school is 375.5. This isn't to say that the highest score that a student received on the test is that number. Rather, that school could've had someone score the full 500 and another student who scored in the double digits. This is important to keep in mind while parsing through the data.

The other measures use various statistical techniques (HETOP and HOMOP, for example) to derive their normalized values, which I did not want to dig much deeper into. I believe using the NAEP score should be sufficient for the purposes of this data analysis.

Another interesting story comes from the "gaps" (disparity in average scores between sub-groups within a year/class/subject)

In [28]:
# Columns which contain values for the gaps between scores of:
# White v Asian students
#       v Black students
#       v Hispanic students
# Male v Female students

gap_cols= ["mn_wag","mn_wbg","mn_whg","mn_mfg"]

desc_long_naep.loc[:,gap_cols]

mn_wag         mn_wbg         mn_whg         mn_mfg
count  84024.000000  164413.000000  196243.000000  658277.000000
mean      -7.432532      20.692708      16.047852      -4.591977
std       14.264807       9.994392      10.275893       8.664576
min     -131.135770     -25.843262     -43.132675    -106.446230
25%      -16.002594      14.246857       9.267326      -9.915894
50%       -8.062675      20.054749      15.276093      -4.178528
75%        0.369026      26.478058      22.158958       1.042725
max      137.154270     147.544370     153.679550      58.579193

Looking at the min and max from the *mn_wag* column, it looks like there is a vast disparity between the scores of White students and Asian students (min: -131.1, max: 137.2). In fact, this suggests the opposite.

These scores are derived by subtracting the first subgroup ("W") from the second ("A"). So it appears that in at least one school, Asian student scores are __on average higher__ than their White peers by 131 points. The converse is true given the max value as well.

Armed with this information, we can see that there does indeed appear to be a disparity between White and Black students' as well as White and Hispanic students' scores. The same holds for Male v Female scores, with the disparity favoring Female students.

# Produce Datasets

Derived from the NAEP dataset

* Demographic Inclusive
    * ELA
    * Math
* Demographic Exclusive
    * ELA
    * Math

### Demo ELA

In [29]:
demo_ela= long_naep.loc[long_naep["subject"] == "ela"].dropna(how="any")

In [30]:
demo_ela.head()

leaidC      leaname  fips stateabb  grade  year subject     mn_all  \
172  100007  Hoover City     1       AL      3  2009     ela  214.54395   
176  100007  Hoover City     1       AL      3  2011     ela  217.66377   
178  100007  Hoover City     1       AL      3  2012     ela  215.21539   
180  100007  Hoover City     1       AL      3  2013     ela  213.73804   
182  100007  Hoover City     1       AL      3  2014     ela  219.03290   

        mn_asn     mn_blk     mn_fem     mn_hsp     mn_mal     mn_wht  \
172  220.27069  190.93071  217.31458  195.04224  212.36569  224.37381   
176  222.10507  193.69196  219.95566  189.55688  214.78072  231.69139   
178  220.94724  191.52635  219.43379  185.46619  211.59489  226.93578   
180  220.89682  196.55606  215.47179  184.31003  211.50256  224.43622   
182  229.80255  197.46252  222.84422  190.68884  215.92966  228.89838   

       mn_wag     mn_wbg     mn_whg    mn_mfg  
172  4.103119  33.443100  29.331573 -4.948883  
176  9.586319  37.999435  42.134506 -5.174942  
178  5.988541  35.409424  41.469589 -7.838898  
180  3.539398  27.880157  40.126190 -3.969223  
182 -0.904175  31.435852  38.209534 -6.914566

### Demo Math

In [31]:
demo_math= long_naep.loc[long_naep["subject"] == "math"].dropna(how="any")

In [32]:
demo_math.head()

leaidC      leaname  fips stateabb  grade  year subject     mn_all  \
173  100007  Hoover City     1       AL      3  2009    math  222.00957   
177  100007  Hoover City     1       AL      3  2011    math  223.60043   
179  100007  Hoover City     1       AL      3  2012    math  223.30418   
181  100007  Hoover City     1       AL      3  2013    math  226.40253   
183  100007  Hoover City     1       AL      3  2014    math  230.91217   

        mn_asn     mn_blk     mn_fem     mn_hsp     mn_mal     mn_wht  \
173  228.68329  205.59877  220.91368  200.90639  222.98395  228.69203   
177  231.37659  207.22340  222.21732  208.76343  225.15512  231.22810   
179  234.07726  205.46187  223.13036  201.37108  223.36121  229.63614   
181  245.31400  213.29327  225.05655  210.97031  227.76344  231.85593   
183  244.89435  216.53716  230.94514  218.20084  230.68436  236.18112   

        mn_wag     mn_wbg     mn_whg    mn_mfg  
173   0.008743  23.093262  27.785645  2.070267  
177  -0.148483  24.004700  22.464676  2.937805  
179  -4.441116  24.174271  28.265060  0.230850  
181 -13.458069  18.562653  20.885620  2.706894  
183  -8.713226  19.643967  17.980286 -0.260788

### All ELA

In [33]:
all_ela= long_naep.loc[long_naep["subject"] == "ela"]

In [34]:
all_ela.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   
6  100005        Albertville City     1       AL      3  2009     ela   
8  100005        Albertville City     1       AL      3  2010     ela   

      mn_all  mn_asn  mn_blk     mn_fem     mn_hsp     mn_mal     mn_wht  \
0  209.40759     NaN     NaN        NaN        NaN  207.21269        NaN   
2  230.81302     NaN     NaN        NaN        NaN  229.81973        NaN   
4  227.06160     NaN     NaN        NaN        NaN  224.74576        NaN   
6  204.40581     NaN     NaN  206.97821  195.20105  201.54659  211.53754   
8  206.01045     NaN     NaN        NaN  196.38988        NaN  211.12666   

   mn_wag  mn_wbg     mn_whg    mn_mfg  
0     NaN     NaN        NaN       NaN  
2     NaN     NaN        NaN       NaN  
4     NaN     NaN        NaN       NaN  
6     NaN     NaN  16.336487 -5.431625  
8     NaN     NaN  14.736786       NaN

### All Math

In [35]:
all_math= long_naep.loc[long_naep["subject"] == "math"]

In [36]:
all_math.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
1  100002  Alabama Youth Services     1       AL      8  2009    math   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
5  100002  Alabama Youth Services     1       AL      8  2012    math   
7  100005        Albertville City     1       AL      3  2009    math   
9  100005        Albertville City     1       AL      3  2010    math   

      mn_all  mn_asn     mn_blk     mn_fem     mn_hsp     mn_mal     mn_wht  \
1  217.34308     NaN  208.98753        NaN        NaN  215.39435        NaN   
3  221.81458     NaN        NaN        NaN        NaN  223.95609        NaN   
5  229.63074     NaN        NaN        NaN        NaN  229.88904        NaN   
7  216.63632     NaN        NaN  216.87672  211.63123  215.30283  220.40680   
9  228.81721     NaN        NaN        NaN  223.30786        NaN  233.30249   

   mn_wag  mn_wbg    mn_whg    mn_mfg  
1     NaN     NaN       NaN       NaN  
3     NaN     NaN       NaN       NaN  
5     NaN     NaN       NaN       NaN  
7     NaN     NaN  8.775574 -1.573898  
9     NaN     NaN  9.994629       NaN

## Save to CSV

In [37]:
# demo_ela.to_csv(r"resources\datasets\demo_ela.csv")
# demo_math.to_csv(r"resources\datasets\demo_math.csv")
# all_ela.to_csv(r"resources\datasets\all_ela.csv")
# all_math.to_csv(r"resources\datasets\all_math.csv")

# Investigating data integrity at different levels

In [38]:
print(len(demo_ela))
print(len(demo_math))
print(len(all_ela))
print(len(all_math))

27516
26423
424550
401266


In [39]:
col= "stateabb"

print(len(demo_ela[col].value_counts()))
print(len(demo_math[col].value_counts()))
print(len(all_ela[col].value_counts()))
print(len(all_math[col].value_counts()))

50
50
51
51


In [40]:
long_naep.loc[long_naep["stateabb"] == "HI"]

leaidC                         leaname  fips stateabb  grade  year  \
126610  1500030  Hawaii Department Of Education    15       HI      3  2009   
126611  1500030  Hawaii Department Of Education    15       HI      3  2009   
126612  1500030  Hawaii Department Of Education    15       HI      3  2010   
126613  1500030  Hawaii Department Of Education    15       HI      3  2010   
126614  1500030  Hawaii Department Of Education    15       HI      3  2011   
126615  1500030  Hawaii Department Of Education    15       HI      3  2011   
126616  1500030  Hawaii Department Of Education    15       HI      3  2012   
126617  1500030  Hawaii Department Of Education    15       HI      3  2012   
126618  1500030  Hawaii Department Of Education    15       HI      3  2013   
126619  1500030  Hawaii Department Of Education    15       HI      3  2013   
126620  1500030  Hawaii Department Of Education    15       HI      3  2014   
126621  1500030  Hawaii Department Of Education    15       HI      3  2014   
126622  1500030  Hawaii Department Of Education    15       HI      3  2015   
126623  1500030  Hawaii Department Of Education    15       HI      3  2015   
126624  1500030  Hawaii Department Of Education    15       HI      4  2009   
126625  1500030  Hawaii Department Of Education    15       HI      4  2009   
126626  1500030  Hawaii Department Of Education    15       HI      4  2010   
126627  1500030  Hawaii Department Of Education    15       HI      4  2010   
126628  1500030  Hawaii Department Of Education    15       HI      4  2011   
126629  1500030  Hawaii Department Of Education    15       HI      4  2011   
126630  1500030  Hawaii Department Of Education    15       HI      4  2012   
126631  1500030  Hawaii Department Of Education    15       HI      4  2012   
126632  1500030  Hawaii Department Of Education    15       HI      4  2013   
126633  1500030  Hawaii Department Of Education    15       HI      4  2013   
126634  1500030  Hawaii Department Of Education    15       HI      4  2014   
126635  1500030  Hawaii Department Of Education    15       HI      4  2014   
126636  1500030  Hawaii Department Of Education    15       HI      4  2015   
126637  1500030  Hawaii Department Of Education    15       HI      4  2015   
126638  1500030  Hawaii Department Of Education    15       HI      5  2009   
126639  1500030  Hawaii Department Of Education    15       HI      5  2009   
...         ...                             ...   ...      ...    ...   ...   
126664  1500030  Hawaii Department Of Education    15       HI      6  2015   
126665  1500030  Hawaii Department Of Education    15       HI      6  2015   
126666  1500030  Hawaii Department Of Education    15       HI      7  2009   
126667  1500030  Hawaii Department Of Education    15       HI      7  2009   
126668  1500030  Hawaii Department Of Education    15       HI      7  2010   
126669  1500030  Hawaii Department Of Education    15       HI      7  2010   
126670  1500030  Hawaii Department Of Education    15       HI      7  2011   
126671  1500030  Hawaii Department Of Education    15       HI      7  2011   
126672  1500030  Hawaii Department Of Education    15       HI      7  2012   
126673  1500030  Hawaii Department Of Education    15       HI      7  2012   
126674  1500030  Hawaii Department Of Education    15       HI      7  2013   
126675  1500030  Hawaii Department Of Education    15       HI      7  2013   
126676  1500030  Hawaii Department Of Education    15       HI      7  2014   
126677  1500030  Hawaii Department Of Education    15       HI      7  2014   
126678  1500030  Hawaii Department Of Education    15       HI      7  2015   
126679  1500030  Hawaii Department Of Education    15       HI      7  2015   
126680  1500030  Hawaii Department Of Education    15       HI      8  2009   
126681  1500030  Hawaii Department Of Education    15       HI      8  2009   
126682  1500030  Hawaii Department Of Education    15      

In [41]:
col= "stateabb"

stateabb_df= pd.DataFrame({
    "demo_ela": demo_ela[col].value_counts(),
    "demo_math": demo_math[col].value_counts(),
    "all_ela": all_ela[col].value_counts(),
    "all_math": all_math[col].value_counts()
})

# print(demo_ela[col].value_counts())
# print(demo_math[col].value_counts())
# print(all_ela[col].value_counts())
# print(all_math[col].value_counts())

stateabb_df

demo_ela  demo_math  all_ela  all_math
AK      96.0      102.0     1302      1299
AL     227.0      239.0     5489      5500
AR     206.0      206.0     9567      8859
AZ    1008.0     1011.0     6021      6035
CA    4615.0     3335.0    26362     19063
CO     250.0      267.0     2342      2277
CT     590.0      630.0     5375      5367
DC      33.0       40.0       41        41
DE     196.0      224.0      649       650
FL     887.0      781.0     2865      2471
GA     737.0      988.0     7517      7518
HI      42.0       42.0       42        42
IA     291.0      316.0    12627     12618
ID      61.0       77.0     3075      3084
IL    1201.0     1241.0    28006     28065
IN     378.0      494.0    12037     12045
KS     195.0      258.0     8000      8040
KY     180.0      233.0     7142      7142
LA     323.0      366.0     3042      3042
MA     628.0      650.0    10415     10439
MD     453.0      518.0      922       923
ME      25.0       33.0     4454      4369
MI     563.0      733.0    20686     20722
MN    1014.0     1035.0    12699     12711
MO     587.0      604.0    16083     14937
MS     124.0      124.0     5910      5901
MT      12.0       13.0     3056      3076
NC     794.0      868.0     4840      4840
ND      22.0       29.0     2386      2381
NE     124.0      118.0     5599      4666
NH      44.0       47.0     4613      4613
NJ    1753.0     1776.0    16327     16328
NM     129.0      129.0     2797      2790
NV      61.0       62.0      457       459
NY     966.0      999.0    20331     20142
OH     531.0      565.0    25224     24623
OK     484.0      475.0    15034     14374
OR     251.0      291.0     5387      5024
PA     910.0      982.0    20845     20853
RI     113.0      122.0     1259      1277
SC     537.0      570.0     3523      3523
SD      27.0       35.0     3346      3350
TN     378.0      436.0     5584      5457
TX    2793.0     2274.0    34805     28211
UT     314.0      278.0     1601      1401
VA     885.0      318.0     5525      1835
VT       NaN        NaN     3720      3721
WA     848.0      857.0     6459      6457
WI     609.0      613.0    15823     15816
WV      15.0       13.0     2324      2048
WY       6.0        6.0     1015       841

In [42]:
col= "grade"

print(demo_ela[col].value_counts())
print(demo_math[col].value_counts())
print(all_ela[col].value_counts())
print(all_math[col].value_counts())

5    4673
4    4640
3    4614
6    4596
7    4499
8    4494
Name: grade, dtype: int64
3    4942
4    4926
5    4753
6    4653
7    3666
8    3483
Name: grade, dtype: int64
5    71430
4    71310
3    71061
6    70888
7    69956
8    69905
Name: grade, dtype: int64
4    70827
3    70788
5    70074
6    69520
7    61764
8    58293
Name: grade, dtype: int64


In [43]:
col= "leaname"

print(len(demo_ela[col].value_counts()))
print(len(demo_math[col].value_counts()))
print(len(all_ela[col].value_counts()))
print(len(all_math[col].value_counts()))

1202
1210
11723
11699


In [44]:
col= "year"

print(demo_ela[col].value_counts())
print(demo_math[col].value_counts())
print(all_ela[col].value_counts())
print(all_math[col].value_counts())

2012    4933
2013    4417
2011    4241
2010    4124
2009    3859
2015    3170
2014    2772
Name: year, dtype: int64
2012    4367
2013    4234
2011    4074
2010    3919
2009    3563
2015    3446
2014    2820
Name: year, dtype: int64
2012    64590
2011    64478
2010    64310
2013    64018
2009    63786
2015    52945
2014    50423
Name: year, dtype: int64
2011    62424
2009    61490
2010    61103
2012    60618
2013    59421
2015    49473
2014    46737
Name: year, dtype: int64


# New DF restrictions
Restrict data to all schools in 2015.

In [49]:
ela2015_df= long_naep.loc[(long_naep["subject"] == "ela") & (long_naep["year"] == 2015)]
math2015_df= long_naep.loc[(long_naep["subject"] == "math") & (long_naep["year"] == 2015)]

In [50]:
print(len(ela2015_df))
print(len(math2015_df))

52945
49473


In [51]:
print(len(ela2015_df["leaname"].value_counts()))
print(len(math2015_df["leaname"].value_counts()))

9512
9464


In [52]:
ela2015_df.to_csv("resources/datasets/ela2015.csv")
math2015_df.to_csv("resources/datasets/math2015.csv")